### PR TITLE
[FLINK-14355] [docs] Example code in state processor API docs doesn't compile

### PR DESCRIPTION
**Pull request by buptljy**
Fix compiling errors in examples of state processor api.

### CODE REVIEW

File: docs/dev/libs/state_processor_api.md [hunk 1]
```diff
@@ -238,9 +238,11 @@ This means if an operator contains a stateful process function such as:
 public class StatefulFunctionWithTime extends KeyedProcessFunction<Integer, Integer, Void> {
  
    ValueState<Integer> state;
- 
+   ValueStateDescriptor<Integer> stateDescriptor;
+   
    @Override
    public void open(Configuration parameters) {
+	  stateDescriptor = new ValueStateDescriptor<>("state", Types.INT);
```
Comment (KarmaGYZ) [inline docs/dev/libs/state_processor_api.md]: Just use whitespace here.

File: docs/dev/libs/state_processor_api.md [hunk 2]
```diff
@@ -256,9 +258,11 @@ public class StatefulFunctionWithTime extends KeyedProcessFunction<Integer, Inte
 public class StatefulFunctionWithTime extends KeyedProcessFunction[Integer, Integer, Void] {
  
   var state: ValueState[Integer];
+  var stateDescriptor: ValueStateDescriptor[Integer];
  
    @throws[Exception]
```
Comment (klion26) [inline docs/dev/libs/state_processor_api.md]: Just curious, why we add `@throws[Exception]` here, where will the Exception throw from?

File: docs/dev/libs/state_processor_api.md (hunk 1 above)
Comment (klion26) [inline docs/dev/libs/state_processor_api.md]: Nit: can we align these two lines.
In other word, do we need to align all the StateDescriptor creating?

File: docs/dev/libs/state_processor_api.md [hunk 3]
```diff
@@ -256,9 +258,11 @@ public class StatefulFunctionWithTime extends KeyedProcessFunction<Integer, Inte
 public class StatefulFunctionWithTime extends KeyedProcessFunction[Integer, Integer, Void] {
  
   var state: ValueState[Integer];
+  var stateDescriptor: ValueStateDescriptor[Integer];
```
Comment (sjwiesman) [inline docs/dev/libs/state_processor_api.md]: This does not need to be a member variable, just declare it inside of open.

File: docs/dev/libs/state_processor_api.md [hunk 4]
```diff
@@ -310,9 +316,11 @@ DataSet<KeyedState> keyedState = savepoint.readKeyedState("my-uid", new ReaderFu
 case class KeyedState(key: Int, value: Int)
  
 class ReaderFunction extends KeyedStateReaderFunction[Integer, KeyedState] {
-  var state ValueState[Integer];
+  var state: ValueState[Integer];
+  var stateDescriptor: ValueStateDescriptor[Integer]
```
Comment (sjwiesman) [inline docs/dev/libs/state_processor_api.md]: Same, just declare this variable inside of open.

**Comment by buptljy** [comment]
@sjwiesman @klion26 @KarmaGYZ  I've addressed your comments. Could you take a look again?

**Pull request merged**